### PR TITLE
Keep report minified when verifying one

### PR DESCRIPTION
**Pull request by m5r**
# Description

Pull the document data from the database record instead of the app's global store when setting the verification fields

medic/cht-core#6802

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/ts/effects/reports.effects.ts [hunk 1]
```diff
@@ -334,9 +334,15 @@ export class ReportsEffects {
             .get(report.doc._id)
             .then(existingRecord => {
               this.reportActions.setFirstSelectedReportDocProperty({ _rev: existingRecord._rev });
```
Comment (dianabarsan) [inline webapp/src/ts/effects/reports.effects.ts:336]: I don't think this line is necessary anymore. It was useful before when we actually pushed the doc we had in the state, but right now it just populates the state with a rev that's about to become irrelevant once we put in the next line. 
Also there's another line in the same file where we only minify the contact lineage. That can also be removed.

File: webapp/src/ts/effects/reports.effects.ts [hunk 2]
```diff
@@ -334,9 +334,15 @@ export class ReportsEffects {
...
+              const doc = getFirstSelectedReport().doc;
+              const nextRecord = {
```
Comment (dianabarsan) [inline webapp/src/ts/effects/reports.effects.ts:338]: Please call this `verifiedRecord` or something more suggestive.

File: webapp/tests/karma/ts/effects/reports.effects.spec.ts [hunk 3]
```diff
@@ -894,7 +894,7 @@ describe('Reports effects', () => {
 
       sinon.stub(Date, 'now').returns(1000); // using faketimers breaks fakeAsync's tick :(
       dbService.put.resolves();
-      dbService.get.resolves({ _id: 'report', _rev: 3 });
+      dbService.get.resolves({ _id: 'report', _rev: 3, contact: { _id: 'contact', parent: { _id: 'parent' } } });
```
Comment (dianabarsan) [inline webapp/tests/karma/ts/effects/reports.effects.spec.ts:897]: Can you please update this test so the doc in the state is hydrated?

**Review by dianabarsan — CHANGES_REQUESTED** [review]
Good quick work on this! 
I left a couple of comments inline. Also, please add some assertions in this test https://github.com/medic/cht-core/blob/master/tests/e2e/forms/add-family-form.specs.js#L57 so we check this end-to-end. 
Thanks!

File: webapp/src/ts/effects/reports.effects.ts (hunk 1 above)
Comment (m5r) [reply webapp/src/ts/effects/reports.effects.ts:336]: I wasn't sure if it still made sense to keep this line so thanks for clarifying :+1:

**Review by dianabarsan — COMMENTED** [review]
I'm adding this comment so I don't get reminded about this review every 30 minutes :D as some changes are still pending.

**Review by dianabarsan — COMMENTED** [review]
Great work on the e2e test! Thanks! I left a few suggestions inline.

**Comment by m5r** [comment]
Thanks a lot for your suggestions @dianabarsan!

**Review by dianabarsan — APPROVED** [review]
Very nice <3

**Pull request merged**